QuizApp/Scenes/HomePage/ViewController/ColorfulBackgroundView.swift

```diff
@@ -0,0 +1,204 @@
+    }
+    
+    private func setupViewHierarchy() {
+        addSubview(mainStackView)
+        gpaView.addSubview(gpaStackView)
+        gpaStackView.addArrangedSubviews(
+            gpaLabel,
+            gpaScoreLabel
+        )
+        mainStackView.addArrangedSubviews(
+            gpaView,
+            detailsButton
+        )
+    }
+    
+    private func setConstraints() {
+        setMainStackViewConstraints()
+        setGpaStackViewConstraints()
+    }
+    
+    private func setViewConstraints() {
+        NSLayoutConstraint.activate([
+            self.heightAnchor.constraint(equalToConstant: Constants.Sizing.height)
+        ])
+    }
+    
+    private func setMainStackViewConstraints() {
+        NSLayoutConstraint.activate(
+            [
+                mainStackView.topAnchor.constraint(
+                    equalTo: topAnchor,
+                    constant: Constants.Sizing.mainStackViewVerticalPadding
+                ),
+                mainStackView.leadingAnchor.constraint(
+                    equalTo: leadingAnchor,
+                    constant: Constants.Sizing.mainStackViewVerticalPadding
+                ),
+                mainStackView.trailingAnchor.constraint(
+                    equalTo: trailingAnchor,
+                    constant: -Constants.Sizing.mainStackViewSidePadding
+                ),
+                mainStackView.bottomAnchor.constraint(
+                    equalTo: bottomAnchor,
+                    constant: -Constants.Sizing.mainStackViewVerticalPadding
+                )
+            ]
+        )
+    }
+    
+    private func setGpaStackViewConstraints() {
+        NSLayoutConstraint.activate(
+            [
+                gpaStackView.topAnchor.constraint(
+                    equalTo: gpaView.topAnchor,
+                    constant: Constants.Sizing.gpaStackViewVerticalPadding
+                ),
+                gpaStackView.bottomAnchor.constraint(
+                    equalTo: gpaView.bottomAnchor,
+                    constant: -Constants.Sizing.gpaStackViewVerticalPadding
+                ),
+                gpaStackView.leadingAnchor.constraint(
+                    equalTo: gpaView.leadingAnchor,
+                    constant: Constants.Sizing.gpaStackViewSidePadding
+                ),
+                gpaStackView.trailingAnchor.constraint(
+                    equalTo: gpaView.trailingAnchor,
+                    constant: -Constants.Sizing.gpaStackViewSidePadding
+                )
+            ]
+        )
+    }
+    
+    // MARK: - Configure View
+    func configure(gpa: String) {
+        gpaScoreLabel.text = gpa
+    }
+    
+    // MARK: - Add Targets
+    private func addTargets() {
+        detailsButton.addTarget(
+            self,
+            action: #selector(detailsButtonTapped),
+            for: .touchUpInside
+        )
+    }
+    
+    // MARK: - Actions
+    @objc private func detailsButtonTapped() {
```

Comment:
Please use modern approach regarding the actions for buttons

Reply:
I prefer this but thanks, will use in the future for sure!